### ISSUE TITLE
Counter signature Regression

### ISSUE THREAD

**Issue by st-elo** [openssl/openssl#12576]
This issue was created after discussion on https://github.com/openssl/openssl/issues/8923#issuecomment-666325543.
According to RFC-5652 (https://tools.ietf.org/html/rfc5652#section-11.4):
>   Countersignature attribute values have ASN.1 type Countersignature:
>
>      Countersignature ::= SignerInfo
>    Countersignature values have the same meaning as SignerInfo values
>    for ordinary signatures, except that:
>    1.  **The signedAttributes field MUST NOT contain a content-type
>        attribute; there is no content type for countersignatures.**
>    2.  The signedAttributes field MUST contain a message-digest
>        attribute if it contains any other attributes.
>    3.  The input to the message-digesting process is the contents octets
>        of the DER encoding of the signatureValue field of the SignerInfo
>        value with which the attribute is associated.

But the patch #8944 made content-type attribute mandatory when calling CMS_SignerInfo_sign for counter-signature.
So conditions of CMS_si_check_attributes should be changed; or its call should be conditioned to whether or not the SignerInfo is for a counter signature.

This behavior was not present in OpenSSL 1.0.2t and has been observed in (at least) 1.1.1g.

**Comment by slontis** [comment]
@st-elo Would you be able to provide some openssl test or sample code so that this scenario could be added to a test.
I am not going to be able to address this for a while since I have higher 3.0 priorities. The change itself shouldnt be that hard.
I can review if you want to add a PR. (i.e: look in the unsigned attributes for a countersignature - if there is one then use a different table for the check. Add a second table for the countersignature case. The biggest change would be adding a test)

**Comment by mspncp** [comment]
> @st-elo Would you be able to provide some openssl test or sample code so that this scenario could be added to a test.
> I am not going to be able to address this for a while since I have higher 3.0 priorities. The change itself shouldnt be that hard.
> I can review if you want to add a PR. (i.e: look in the unsigned attributes for a countersignature - if there is one then use a different table for the check. Add a second table for the countersignature case. The biggest change would be adding a test)

ping @st-elo

**Comment by st-elo** [comment]
Yes ! I'll do so ASAP. I'll also give the workaround we use, which lowers controls on content-type attribute for all signerinfo.

**Comment by nhorman** [comment]
Marking as inactive, to be closed at the end of 3.4 dev barring further input

**Comment by t8m** [comment]
Closing as the issue was labeled as inactive for extended period of time and there was no additional input.
Please add a comment if you have any further information to support reopening the issue.
